### DISCUSSION TITLE
Integration with LLVM Clang on MacOS

### DISCUSSION THREAD

**Discussion by alexolog** [ccache/ccache#1269] [Questions & answers]
I have Ccache installed on an M1 Mac, currently using the "native" Apple Clang from Xcode. I'd like to install LLVM 16 from homebrew and use it instead. What do I need to do to make Ccache work with the LLVM Clang?

Thank you!

**Comment by jrosdahl** [comment]
In one way or another, Xcode needs to call ccache using one of the methods described in the _[Run modes](https://ccache.dev/manual/4.8.html#_run_modes)_ part of the Ccache documentation.

**Reply by alexolog** [reply]
That's the thing, I don't want to use Xcode.
If I do the following
```
cp ccache /usr/local/bin/
ln -s ccache /usr/local/bin/gcc
```
How will ccache know to run the LLVM compiler instead of the Xcode one?

**Reply by jrosdahl** [reply]
The key part is this:

> This will work as long as the directory with symbolic links or ccache copies comes before the directory with the compiler (typically /usr/bin) in PATH.

That is, if you do

```
cp ccache /usr/local/bin
ln -s ccache /usr/local/bin/clang
```

and you have `/usr/local/bin` early in your `PATH` so that your shell finds `/usr/local/bin/clang` before `/whereever/llvm/bin/clang`, then ccache will be able to find `/whereever/llvm/bin/clang` (assuming that `/whereever/llvm/bin` is also in `PATH` but later than `/usr/local/bin`).

**Comment by alexolog** [comment]
So ccache just searches the path?

**Reply by jrosdahl** [reply]
Yes.
